### PR TITLE
Desc Keys Tests 2

### PR DESCRIPTION
**Pull request by andytill**
Tests for descending keys in Riak Time Series.

### CODE REVIEW

**Comment by hazen** [comment]
🎉  Looks much better this time around

**Comment by andytill** [comment]
@javajolt copy/paste FTW!

**Comment by hazen** [comment]
Looks like `ts_cluster_keys_SUITE` and `ts_simple_single_key_ops` pass, but `ts_cluster_select_desc_SUITE` and `ts_cluster_stream_list_keys_SUITE` have test failures. I'll need to crack the HTML of common test to find out why.

Looks like there are still some `ts_util` functions.  Finding in the code...

File: tests/ts_cluster_select_desc_SUITE.erl [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+%%--------------------------------------------------------------------
+%% COMMON TEST CALLBACK FUNCTIONS
+%%--------------------------------------------------------------------
+
+suite() ->
+    [{timetrap,{minutes,10}}].
+
+init_per_suite(Config) ->
+    [_Node|_] = Cluster = ts_util:build_cluster(multiple),
```
Comment (hazen) [inline tests/ts_cluster_select_desc_SUITE.erl]: Looks like it should be `ts_setup:build_cluster(3)`

File: tests/ts_cluster_stream_list_keys_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+%%--------------------------------------------------------------------
+%% COMMON TEST CALLBACK FUNCTIONS
+%%--------------------------------------------------------------------
+
+suite() ->
+    [{timetrap,{minutes,10}}].
+
+init_per_suite(Config) ->
+    [Node|_] = Cluster = ts_util:build_cluster(multiple),
```
Comment (hazen) [inline tests/ts_cluster_stream_list_keys_SUITE.erl]: Looks like it should be `ts_setup:build_cluster(3)`

File: tests/ts_cluster_stream_list_keys_SUITE.erl [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+%%--------------------------------------------------------------------
+%% COMMON TESTS
+%%--------------------------------------------------------------------
+
+%%%
+%%%
+%%%
+
+table_def_basic_create_data(Cluster, Pid) ->
+    ts_util:create_and_activate_bucket_type(Cluster,
```
Comment (hazen) [inline tests/ts_cluster_stream_list_keys_SUITE.erl]: Needs to be broken into

```
    ts_setup:create_bucket_type(Cluster, DDL, Table),
    ts_setup:activate_bucket_type(Cluster, Table),
```

File: tests/ts_cluster_stream_list_keys_SUITE.erl [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        [{1,1,N} || N <- lists:seq(100,5000,100)],
+        stream_list_keys_receive(ReqId)
+    ).
+
+%%%
+%%%
+%%%
+
+create_data_in_basic_varchar_table(Cluster, Pid) ->
+    ts_util:create_and_activate_bucket_type(Cluster,
```
Comment (hazen) [inline tests/ts_cluster_stream_list_keys_SUITE.erl]: Needs to be broken into

```
    ts_setup:create_bucket_type(Cluster, DDL, Table),
    ts_setup:activate_bucket_type(Cluster, Table),
```

**Review by hazen — CHANGES_REQUESTED** [review]
Once these have been migrated away from `ts_util`, they might pass

File: tests/ts_cluster_select_desc_SUITE.erl (hunk 1 above)
Comment (andytill) [reply tests/ts_cluster_select_desc_SUITE.erl]: Fixed in 619abbe

File: tests/ts_cluster_stream_list_keys_SUITE.erl (hunk 2 above)
Comment (andytill) [reply tests/ts_cluster_stream_list_keys_SUITE.erl]: Fixed in 619abbe

File: tests/ts_cluster_stream_list_keys_SUITE.erl (hunk 3 above)
Comment (andytill) [reply tests/ts_cluster_stream_list_keys_SUITE.erl]: Fixed in 619abbe

File: tests/ts_cluster_stream_list_keys_SUITE.erl (hunk 4 above)
Comment (andytill) [reply tests/ts_cluster_stream_list_keys_SUITE.erl]: Fixed in 619abbe

**Comment by andytill** [comment]
@javajolt  I had been compiling with an old `ts_util.beam` file in my ebin directory without realising it, so the tests were passing fine locally. Everything has been updated to use the new utils. Great review and comments, thanks!

**Review by hazen — APPROVED** [review]

File: tests/ts_cluster_select_desc_SUITE.erl [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+            register(random_num_proc, self()),
+            random:seed(),
+            random_num_proc()
+        end),
+    [{cluster, Cluster} | Config].
+
+end_per_suite(_Config) ->
+    ok.
+
+init_per_group(use_ttb_true, Config) ->
```
Comment (hazen) [inline tests/ts_cluster_select_desc_SUITE.erl:47]: Love that you are testing both of these cases

**Comment by hazen** [comment]
:+1: 619abbe

File: tests/ts_cluster_select_desc_SUITE.erl (hunk 5 above)
Comment (andytill) [reply tests/ts_cluster_select_desc_SUITE.erl:47]: @javajolt  Thanks, I meant to show you at Burlington but I was hustling DDL upgrades before my hols. We could trivially apply this to some other suites.

**Pull request merged**